server/app/parsers/ProgramMigrationParser.java

```diff
@@ -0,0 +1,20 @@
+package parsers;
+
+import javax.inject.Inject;
+import play.http.DefaultHttpErrorHandler;
+import play.mvc.BodyParser;
+
+/**
+ * This is a custom HTTP request body parser for Program Migration that increases the allowable
+ * limit for buffering. Program json is often quite large and we want to allow it to be processed
+ * without increasing the buffer limit for all HTTP requests. All other requests use the default
+ * Play buffer limit which is 100KB. For more information see:
+ * https://www.playframework.com/documentation/3.0.x/JavaBodyParsers
+ */
+public class ProgramMigrationParser extends BodyParser.FormUrlEncoded {
```

Comment:
(blocking) There's nothing specific about this for program migration. Can you:
- move the program migration specific comments to the `hxImportProgram` controller action 
- rename this to something like `LargeBodyParser` or `LargeFormUrlEncodedBodyParser`
- include in the javadoc that this supports 1 MiB (mebibyte)

Reply:
done!